aten/src/ATen/native/native_functions.yaml

```diff
@@ -3655,6 +3655,11 @@
     QuantizedCUDA: quantized_max_pool2d_cudnn
   autogen: quantized_max_pool2d.out
 
+- func: quantized_max_pool3d(Tensor self, int[3] kernel_size, int[3] stride=[], int[3] padding=0, int[3] dilation=1, bool ceil_mode=False) -> Tensor
```

Comment:
On the other hand, there are already `quantized_max_pool2d` and `quantized_max_pool1d`. Adding `quantized_max_pool3d` in parallel seems also consistent with them.

Reply:
hmm that shouldn't @bdhirsh are the Quantized key handled differently here?

Reply:
hmm @albanD I think I know what's going on. With the per-backend-functionality dispatch key changes, a tensor with the `QuantizedCPU` dispatch key automatically gets the `AutogradCPU` dispatch key too ([code](https://github.com/pytorch/pytorch/blob/ac1cf00085b30eadd164ccf02e5208a59ec1b38b/c10/core/DispatchKeySet.h#L809))

So what we care about is "what gets registered to the AutogradCPU key" in the dispatcher. But... we want the composite kernel to be registered to the AutogradCPU key! Since the normal CPU backend should run that decomp.

I think there are 3 options:

(1) Add this new quantized-specific op

(2) Try to get the quantized backends to use the AutogradOther dispatch key again, instead of using AutogradCPU. I think that this would fix the problem? But I'd need to think more about what the blast radius of that change would be. wdyt?

(3) Ask users to run this quantized logic under `torch.inference_mode()`. I think this should work (in that it would successfully prevent the autograd key from running, so we hit the quantized kernel instead of the decomp), but this might not be a reasonable ask, since it means you'll get wrong results if you forget to use inference mode.

Reply:
Thanks for the comments. I think option1 as _Add this new quantized-specific op_ is what we used to do in similar kernels of `quantized_max_pool2d` and `quantized_max_pool1d`. @albanD Do you think is it ok we keep this PR in this similar way until option2 is implemented?